### PR TITLE
add io.github.hrkfdn.ncspot

### PR DESCRIPTION
**Pull request by thorpelawrence**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/hrkfdn/ncspot/issues/997#issuecomment-1335430372
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: desktop-icon.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/misc/ncspot.desktop b/misc/ncspot.desktop
```
Comment (hfiguiere) [inline desktop-icon.patch:1]: no need for a patch. you can just use `desktop-file-edit`

`rename-icon` also take care of it but you don't use it here.

File: io.github.hrkfdn.ncspot.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  </categories>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/hrkfdn/ncspot/main/images/screenshot.png</image>
+    </screenshot>
+  </screenshots>
+  <url type="homepage">https://github.com/hrkfdn/ncspot</url>
+  <releases>
+    <release version="0.11.2" date="2022-10-15"/>
+  </releases>
```
Comment (hfiguiere) [inline io.github.hrkfdn.ncspot.appdata.xml:40]: missing required `content_rating`

File: io.github.hrkfdn.ncspot.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      - cargo --offline build --release --locked
+      - install -Dm 755 target/release/ncspot /app/bin/ncspot
+      - install -Dm 644 misc/ncspot.desktop /app/share/applications/io.github.hrkfdn.ncspot.desktop
+      - install -Dm 644 images/logo.svg /app/share/icons/hicolor/scalable/apps/io.github.hrkfdn.ncspot.svg
+    build-options:
+      env:
+        CARGO_HOME: /run/build/ncspot/cargo
+    post-install:
+      - install -Dm 644 io.github.hrkfdn.ncspot.appdata.xml /app/share/appdata/io.github.hrkfdn.ncspot.appdata.xml
+build-options:
```
Comment (hfiguiere) [inline io.github.hrkfdn.ncspot.yml:37]: build-options and finish-args usually come before the modules.

File: io.github.hrkfdn.ncspot.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  append-path: /usr/lib/sdk/rust-stable/bin
+finish-args:
+  - --share=network
+  - --socket=pulseaudio
+  - --talk-name=org.freedesktop.Notifications
+  # clipboard
+  - --socket=wayland
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline io.github.hrkfdn.ncspot.yml:45]: `--share=ipc` is also required for X11.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build io.github.hrkfdn.ncspot

File: io.github.hrkfdn.ncspot.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+command: ncspot
+modules:
+  - name: ncspot
+    sources:
+      - cargo-sources.json
```
Comment (hfiguiere) [inline io.github.hrkfdn.ncspot.yml:11]: this file is actually missing.

**Comment by thorpelawrence** [comment]
bot, build io.github.hrkfdn.ncspot

**Comment by thorpelawrence** [comment]
bot, build io.github.hrkfdn.ncspot

File: io.github.hrkfdn.ncspot.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2019-2022 Henrik Friedrichsen -->
+<component type="desktop">
+  <id>io.github.hrkfdn.ncspot</id>
+  <launchable type="desktop-id">io.github.hrkfdn.ncspot.desktop</launchable>
+  <metadata_license>FSFAP</metadata_license>
+  <project_license>BSD-2-Clause</project_license>
+  <name>ncspot</name>
+  <summary>Cross-platform ncurses Spotify client written in Rust</summary>
```
Comment (hfiguiere) [inline io.github.hrkfdn.ncspot.appdata.xml:9]:
```suggestion
  <summary>Cross-platform ncurses Spotify client</summary>
```

This doesn't matter to users.

File: io.github.hrkfdn.ncspot.yml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        path: io.github.hrkfdn.ncspot.appdata.xml
+    buildsystem: simple
+    build-commands:
+      - cargo --offline fetch --manifest-path Cargo.toml
+      - cargo --offline build --release --locked
+      - install -Dm 755 target/release/ncspot /app/bin/ncspot
+      - desktop-file-edit --set-key=Icon --set-value=io.github.hrkfdn.ncspot misc/ncspot.desktop
+      - install -Dm 644 misc/ncspot.desktop /app/share/applications/io.github.hrkfdn.ncspot.desktop
+      - install -Dm 644 images/logo.svg /app/share/icons/hicolor/scalable/apps/io.github.hrkfdn.ncspot.svg
+    post-install:
```
Comment (hfiguiere) [inline io.github.hrkfdn.ncspot.yml:47]:
```suggestion
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build io.github.hrkfdn.ncspot

File: io.github.hrkfdn.ncspot.appdata.xml (hunk 6 above)
Comment (hfiguiere) [reply io.github.hrkfdn.ncspot.appdata.xml:9]: it actually has meaning for the end user as it qualifies the UI. Albeit "text based" or "terminal based" is probably more descriptive.

**Comment by barthalion** [comment]
bot, build io.github.hrkfdn.ncspot

**Comment by thorpelawrence** [comment]
> It doesn't launch from CLI:
> 
> ```
> ~ 15.2s ❱ flatpak run io.github.hrkfdn.ncspot
> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 6, kind: Uncategorized, message: "No such device or address" }', /run/build/ncspot/cargo/vendor/cursive-0.20.0/src/cursive_ext.rs:70:36
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```
> 
> Works fine through the desktop file, though.

Interesting, I can't reproduce from the 11183 build - it launches for me through both methods

**Comment by barthalion** [comment]
Hm, looks like something is wrong when I launch it inside the toolbox. LGTM, then.

/merge @hrkfdn

**Pull request closed without merging**